src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -117,18 +119,13 @@ impl CommandRunner {
         Component::Normal(name) => name
```

Comment:
This is pre-existing, but its an un-necessary check since directory_path is a `RelativePath` which already ensures all components are `Normal`:
https://github.com/pantsbuild/pants/blob/09606b45b25861d971a03121bef1e3991670bc30/src/rust/engine/fs/src/lib.rs#L85-L109

Reply:
Yes, but the code still needs to destructure the component to get at the `name`. Is this not the preferred way to do that?